### DISCUSSION TITLE
SetAllocationSizeAsync vs SetEndOfFileAsync

### DISCUSSION THREAD

**Discussion by TonyValenti** [dokan-dev/dokan-dotnet#333] [Q&A] [answered]
What is the difference between these two?  They seem to do the same thing.

**Comment by Liryna** [comment, marked as answer]
SetAllocationSize: https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/0201c69b-50db-412d-bab3-dd97aeede13b
SetEndOfFile: https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/75241cca-3167-472f-8058-a52d77c6bb17

The difference is really between the physical storage space and the real data size inside the physical storage.

**Comment by TonyValenti** [comment]
Thanks so much!

One other question - 
When returning FileInformation, should the file name be relative or absolute? Or does it matter?

**Reply by Liryna** [reply]
Please see the sample https://github.com/dokan-dev/dokan-dotnet/blob/master/sample/DokanNetMirror/Mirror.cs#L736
